packages/back-end/src/services/stats.ts

```diff
@@ -331,3 +336,95 @@ export async function analyzeExperimentResults({
     dimensions,
   };
 }
+
+export function analyzeExperimentTraffic({
+  rows,
+  variations,
+}: {
+  rows: ExperimentAggregateUnitsQueryResponseRows;
+  variations: SnapshotSettingsVariation[];
+}): ExperimentSnapshotTraffic {
+  if (!rows || !rows.length) {
+    return {
+      overall: [],
+      dimension: {},
+      error: "NO_ROWS_IN_UNIT_QUERY",
+    };
+  }
+  if (rows.length == MAX_ROWS_UNIT_AGGREGATE_QUERY) {
+    return {
+      overall: [],
+      dimension: {},
+      error: "TOO_MANY_ROWS",
+    };
+  }
+
+  // build variation data to check traffic
+  const variationIdMap: { [key: string]: number } = {};
+  const variationWeights: number[] = [];
+  variations.forEach((v, i) => {
+    variationIdMap[v.id] = i;
+    variationWeights.push(v.weight);
+  });
+
+  // use nested objects to easily fill values as we iterate over
+  // query result
+  const dimTrafficResults: Map<
+    string,
+    Map<string, ExperimentSnapshotTrafficDimension>
+  > = new Map();
+
+  // instantiate return object here, as we can fill `overall`
+  // unit data on the first pass
+  const trafficResults: ExperimentSnapshotTraffic = {
+    overall: [
+      {
+        name: "All",
+        srm: 0,
+        variationUnits: Array(variations.length).fill(0),
+      },
+    ],
+    dimension: {},
+  };
+
+  rows.forEach((r) => {
+    const variationIndex = variationIdMap[r.variation];
+    const dimTraffic: Map<string, ExperimentSnapshotTrafficDimension> =
+      dimTrafficResults.get(r.dimension_name) ?? new Map();
+    const dimValueTraffic: ExperimentSnapshotTrafficDimension = dimTraffic.get(
+      r.dimension_value
+    ) || {
+      name: r.dimension_value,
+      srm: 0,
+      variationUnits: Array(variations.length).fill(0),
+    };
+    // assumes one row per dimension slice in the payload, use += if there will be multiple
+    dimValueTraffic.variationUnits[variationIndex] = r.units;
+
+    dimTraffic.set(r.dimension_value, dimValueTraffic);
+    dimTrafficResults.set(r.dimension_name, dimTraffic);
+
+    // aggregate over date unit counts for overall unit counts
+    if (r.dimension_name === "dim_exposure_date") {
```

Comment:
Maybe make this a shared const. We're using the same value to access the values for the `UnitCountDateGraph`

Reply:
Yeah I was considering that. We may need to sync later about the dimension names in general.